src/Microsoft.OData.Client/BulkUpdateSaveResult.cs

```diff
@@ -0,0 +1,236 @@
+        /// <summary>Instance of ODataWriter used to write the bulk update request.</summary>
+        private ODataWriterWrapper writer;
+
+        /// <summary>
+        /// Instance of the BulkUpdateGraph that contains descriptors with their 
+        /// related descriptors to be used in writing bulk update requests.
+        /// </summary>
+        private BulkUpdateGraph bulkUpdateGraph;
+
+        #endregion
+
+        /// <summary>
+        /// constructor for BulkUpdateSaveResult
+        /// </summary>
+        /// <param name="context">The runtime context of the data service.</param>
+        /// <param name="method">Method name for the SaveChanges method.</param>
+        /// <param name="options">Options when saving changes.</param>
+        /// <param name="callback">The user callback.</param>
+        /// <param name="state">The user state object.</param>
+        internal BulkUpdateSaveResult(DataServiceContext context, string method, SaveChangesOptions options, AsyncCallback callback, object state)
+            : base(context, method, null, options, callback, state)
+        {
+            Debug.Assert(Util.IsBulkUpdate(options), "the options must have bulk update flag set");
+            this.bulkUpdateGraph = new BulkUpdateGraph(this.RequestInfo);
+        }
+
+        /// <summary>
+        /// Returns an instance of the <see cref="BulkUpdateGraph"/>
+        /// </summary>
+        internal BulkUpdateGraph BulkUpdateGraph
+        {
+            get { return this.bulkUpdateGraph; }
+        }
+
+        protected override Stream ResponseStream => throw new NotImplementedException();
+
+        /// <summary>
+        /// Syncronous bulk update request.
+        /// </summary>
+        /// <typeparam name="T"> The type of the top-level objects to be deep-updated.</typeparam>
+        /// <param name="objects"> The top-level objects of the type to be deep updated.</param>
+        internal void BulkUpdateRequest<T>(params T[] objects)
+        {
+            BuildDescriptorGraph(this.ChangedEntries, true, objects);
+
+            // This call ensures that the top-level descriptors' EntitySetName is set properly before the request processing begins. 
+            this.bulkUpdateGraph.ComputeEntitySetName();
+
+            ODataRequestMessageWrapper bulkUpdateRequestMessage = this.GenerateBulkUpdateRequest();
+
+            if (bulkUpdateRequestMessage != null)
+            {
+                bulkUpdateRequestMessage.SetRequestStream(bulkUpdateRequestMessage.CachedRequestStream);
+
+                try
+                {
+                    this.batchResponseMessage = this.RequestInfo.GetSynchronousResponse(bulkUpdateRequestMessage, false);
+                }
+                catch (DataServiceTransportException ex)
+                {
+                    InvalidOperationException exception = WebUtil.GetHttpWebResponse(ex, ref this.batchResponseMessage);
+
+                    throw exception;
+                }
+            }

```

Comment:
We can avoid unnecessary indentation here by returning early if we detect `null`, rather than wrapping the entire code in a non-null check.

```suggestion
            if (bulkUpdateRequestMessage == null)
            {
                return;
            }

            bulkUpdateRequestMessage.SetRequestStream(bulkUpdateRequestMessage.CachedRequestStream);

            try
            {
                this.batchResponseMessage = this.RequestInfo.GetSynchronousResponse(bulkUpdateRequestMessage, false);
            }
            catch (DataServiceTransportException ex)
            {
                InvalidOperationException exception = WebUtil.GetHttpWebResponse(ex, ref this.batchResponseMessage);

                throw exception;
            }
```